src/test/java/org/opensearch/knn/TestUtils.java

```diff
@@ -32,6 +37,105 @@
     public static final String BWCSUITE_ROUND = "tests.rest.bwcsuite_round";
     public static final String TEST_CLUSTER_NAME = "tests.clustername";
 
+    //Generating index vectors using random function with a seed which makes these vectors standard and generate same vectors for each run.
+    public static float[][] randomlyGenerateStandardIndexVectors(int numVectors, int dimensions) {
+        float[][] standardIndexVectors = new float[numVectors][dimensions];
+        int seedVal = 500;
+
+        for (int i = 0; i < numVectors; i++) {
+            float[] vector = new float[dimensions];
+            for (int j = 0; j < dimensions; j++) {
+                Random random = new Random(seedVal++);
+                vector[j] = random.nextFloat();
+            }
+            standardIndexVectors[i] = vector;
+        }
+        return standardIndexVectors;
+    }
+
+    //Generating query vectors using random function with a seed which makes these vectors standard and generate same vectors for each run.
+    public static float[][] randomlyGenerateStandardQueryVectors(int numVectors, int dimensions){
+        float[][] standardQueryVectors = new float[numVectors][dimensions];
+        int seedVal = 1;
+
+        for (int i = 0; i < numVectors; i++) {
+            float[] vector = new float[dimensions];
+            for (int j = 0; j < dimensions; j++) {
+                Random random = new Random(seedVal++);
+                vector[j] = random.nextFloat();
+            }
+            standardQueryVectors[i] = vector;
+        }
+        return standardQueryVectors;
+    }
+
+    public static float[][] generateRandomVectors(int numVectors, int dimensions){
+        float[][] randomVectors = new float[numVectors][dimensions];
+
+        for (int i = 0; i < numVectors; i++) {
+            float[] vector = new float[dimensions];
+            for (int j = 0; j < dimensions; j++) {
+                vector[j] = random().nextFloat();
+            }
+            randomVectors[i] = vector;
+        }
+        return randomVectors;
+    }
+
+    // Here, for a given space type we will compute the 'k' shortest distances among all the index vectors for each and every query vector.
+    // These computed distances are later used while calculating Recall value to compare with the 'k' index vectors obtained for each and every search query performed.
+    public static float[][] computeGroundTruthDistances(float[][] indexVectors, float[][] queryVectors, SpaceType spaceType, int k){
+        float[][] groundTruthValues = new float[queryVectors.length][k];
```

Comment:
In this approach, we don't need to know the docIds or index Vectors of the shortest k distances computed and stored in GTValues because if the k resulting index vectors obtained from each search query gives the same l2 shortest distances in GT values, then recall value will be 1 else it will be less than 1.

It is a very good catch about using docIds and I agree with you. Will update code.